action/protocol/managers.go

```diff
@@ -33,18 +41,6 @@ func LegacyKeyOption(key hash.Hash160) StateOption {
 	}
 }
 
-// FilterOption sets the filter
-func FilterOption(cond db.Condition, minKey, maxKey []byte) StateOption {
-	return func(cfg *StateConfig) error {
-		cfg.Cond = cond
-		cfg.MinKey = make([]byte, len(minKey))
-		copy(cfg.MinKey, minKey)
-		cfg.MaxKey = make([]byte, len(maxKey))
-		copy(cfg.MaxKey, maxKey)
-		return nil
-	}
-}
-
```

Comment:
let's still keep this option, in case we need better performance, it does not affect the correctness of this fix/PR

Reply:
you can add it back in the PR which it is needed.